src/main/java/de/tum/in/www1/artemis/web/rest/GradingInstructionResource.java

```diff
@@ -0,0 +1,145 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.GradingInstruction;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.ExerciseService;
+import de.tum.in.www1.artemis.service.GradingInstructionService;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * REST controller for managing Structured Grading Instructions.
+ */
+@RestController
+@RequestMapping({ "/api" })
+public class GradingInstructionResource {
+
+    private final Logger log = LoggerFactory.getLogger(GradingInstructionResource.class);
+
+    private static final String ENTITY_NAME = "gradingInstruction";
+
+    @Value("${jhipster.clientApp.name}")
+    private String APPLICATION_NAME;
+
+    private final GradingInstructionService gradingInstructionService;
+
+    private final ExerciseService exerciseService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public GradingInstructionResource(GradingInstructionService gradingInstructionService, ExerciseService exerciseService, AuthorizationCheckService authCheckService) {
+        this.gradingInstructionService = gradingInstructionService;
+        this.exerciseService = exerciseService;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * GET /exercises/:exerciseId/grading-instructions : get the "id" exercise.
+     *
+     * @param exerciseId the id of the exercise to retrieve its grading instructions
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @GetMapping("/exercises/{exerciseId}/grading-instructions")
```

Comment:
Please refactor these endpoint Strings into a public constant subclass `Endpoints`. We already did this for programming exercises (`ProgrammingExerciseResource`) and a part of the `ParticipationResource`. This helps keeping the Strings between tests and resources (and also within the resource itself) consistent.